scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryNetwork.scala

```diff
@@ -193,11 +193,11 @@ object DiscoveryNetwork {
         maybeResponse.attempt.flatMap {
           case Right(Some(response)) =>
             f(response)
-          case Right(None) =>
-            Task.unit
           case Left(NonFatal(ex)) =>
             // Not responding to this one, but it shouldn't stop handling further requests.
             Task(logger.error(s"Error handling incoming request: $ex"))
+          case _ => // Right(None) or fatal? - exhaustive match
+            Task.unit
```

Comment:
Although that's not the same either since it would also handle any errors raised by `f`. Another option would be to add 
```scala
  case ex => 
     Task.raiseError(ex)
```
and call it a day.

Reply:
Last variant :) 
```scala
maybeResponse.onErrorRecoverWith {
  case NonFatal(ex) =>
    // Not responding to this one, but it shouldn't stop handling further requests.
    Task(logger.error(s"Error handling incoming request: $ex")).as(None)
}.flatMap(_.fold(Task.unit)(f))
```

Reply:
Oops, I have missed that temporary `let-it-compile` fix.
Will fix that like you suggested last.